### PR TITLE
Christian Number input

### PR DESCRIPTION
**Pull request by altffyra**
### **Number input feature**
image

New input feature for smaller numbers that emits changed number to parent.

### Features
Selectable direction
Changeable Label
Dynamic sizing linked to maxLength
Error-handling with icon and text colors
Helper-text to guide user
Disable options

### Working
Storybook
Tests 

### Props
  ``label``?: string - Default=''
  ``validationText``?: string  - Default='validation text'
 ``isValid``?: boolean | null  - Default=null
 ``placeholder``?: string | undefined - Default='undefined'
 ``disabled``?: boolean  - Default=false
  ``helpertext``?: string - Default=''
  ``maxLength``?: number - Default=3
  ``maxValue``?: number - Default=999
  ``direction``?: 'horizontal' | 'vertical'  - Default='horizontal'
 ``increment``?: number - Default=1

### CODE REVIEW

**Review by adamfagerstrom — COMMENTED** [review]
Great work Christian! I have some comments about a11y and some logic. However, a great start for the component. Just polishing.

File: src/components/number-input/number-input.vue [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+});
+</script>
+
+<template>
+  <div class="numbers-input__div">
+    <div :class="['topWrapper', props.disabled ? 'disabled' : '']">
+      <h1 class="h3">
+        {{ props.label }}
+      </h1>
+    </div>
```
[the remark this answers is not in the document]
Comment (altffyra) [reply src/components/number-input/number-input.vue:107]: added label

File: src/components/number-input/number-input.vue [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+          props.disabled ? 'disabled' : '',
+        ]"
+      >
+        <div
+          v-if="props.direction == 'horizontal'"
+          class="horizontal__decrementDiv indicator"
+          role="button"
+          @click="decrement"
+        >
+          <AsyncIcon icon="chevron-left" />
```
[the remark this answers is not in the document]
Comment (altffyra) [reply src/components/number-input/number-input.vue:150]: changed calculation to apply more space

File: src/components/number-input/number-input.vue [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+          :value="props.modelValue"
+          class="h5"
+          @input="emitInput($event.target)"
+        >
+
+        <div
+          v-if="props.direction == 'horizontal'"
+          :class="['horizontal__incrementDiv indicator']"
+          role="button"
+          @click="increment"
```
[the remark this answers is not in the document]
Comment (altffyra) [reply src/components/number-input/number-input.vue:173]: added tab index to all interactable elements aswell as key down space to increment/decrement

File: src/components/number-input/number-input.vue [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+const maxLengthRef: Ref = ref(props.maxLength);
+onBeforeMount(() => {
+  if (maxLengthRef.value > 14) maxLengthRef.value = 14;
+});
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function emitInput(e: any) {
+  if (e.value > props.maxValue) {
+    e.value = props.maxValue;
```
[the remark this answers is not in the document]
Comment (altffyra) [reply src/components/number-input/number-input.vue:49]: from an user view i would be annoyed if the value would not emit what i entered and it shows what you cant enter - 
from a developer view i understand that you want it easier to correct errors but the structure of the input is based on what restrictions is put into the module - not dynamically assigned as the lifetime progresses

File: src/styles/font-system.scss [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+.lowercase {
+  text-transform: lowercase;
+}
+
+
+
+@font-face {
+  font-family: SourceSansPro;
+  src: url(./fonts/SourceSansPro-ExtraLight.ttf);
```
[the remark this answers is not in the document]
Comment (altffyra) [reply src/styles/font-system.scss:110]: i will investigate this in the font system branch

File: src/components/number-input/number-input.vue [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+
+type InputProps = {
+  label?: string
+  validationText?: string
+  isValid?: boolean | null
+  placeholder?: string | undefined
+  disabled?: boolean
+  modelValue?: number
+  helpertext?: string
+  maxLength?: number
```
[the remark this answers is not in the document]
Comment (altffyra) [reply src/components/number-input/number-input.vue:19]: this is handled on line 43 but we moved away from these kinds of inputhandleing / console.log becouse it gives errors in the console in vue. i can restructure it if you want!

[the remark this answers is not in the document]
Comment (adamfagerstrom) [reply src/components/number-input/number-input.vue]: Yeah, maybe. I mean if we create a new type and specify it to `maxlength: 1 | 2 | 3 | 4 .... | 14` (but maybe export it and don't have it in the .vue file. Then you would get an error in the editor before compiling it in the browser. If you proceed, you would get errors in the browser aswell, saying it's a TS-Error and the type don't match (But that's ok, cus it's an error, and we don't allow it). Then you can skip the `onBeforeMount` hook.

File: src/components/number-input/number-input.vue [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+  const emitValue = parseInt(e.value.slice(0, maxLengthRef.value));
+
+  if (emitValue !== props.modelValue) {
+    emit('update:modelValue', emitValue);
+  }
+}
+}
+
+const verticalAlignment = props.direction == 'vertical' ? 'verticalStyle' : '';
+const labelRef = ref(props.label+ ' '+ uuidv4());
```
Comment (adamfagerstrom) [inline src/components/number-input/number-input.vue:68]: We could go with only uuidV4 here, as laberef. I think that a space inside a `id` might mess things up. So only uuid will do just fine! :)

File: src/components/number-input/number-input.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamfagerstrom) [reply src/components/number-input/number-input.vue]: Nice. And go with the smaller font aswell 

`Then i think we can go with a smaller font size maybe body 1 as label, and body 2 or subtitle for helper.`

File: src/components/number-input/number-input.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (saradalsman) [reply src/components/number-input/number-input.vue:189]: Works great, add `cursor:pointer` to the arrows as well

File: src/components/number-input/number-input.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (saradalsman) [reply src/components/number-input/number-input.vue]: I agree with Adam here, that we shouldn't override the value in the input and also emit what is actually in the input field. And if the user types too many digits, just leave the rest.

File: src/components/number-input/number-input.vue [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+      <div
+        v-if="props.direction == 'vertical'"
+        :style="{
+          width:
+            props.direction === 'vertical'
+              ? verticalWidthCalculation + 'px'
+              : '',
+        }"
+        :class="[
+          'vertical__incrementDiv',
```
Comment (saradalsman) [inline src/components/number-input/number-input.vue:133]: There's some inconsistency in the namning format for classnames, for example "numbers-input__div", "topWrapper", "vertical__incrementDiv". Would be nicer if they kept to the same style, I think we've talked about BEM before?

File: src/components/number-input/number-input.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (altffyra) [reply src/components/number-input/number-input.vue]: added new interface with import called maxlengthInterface
maxlength?: maxlengthInterface['maxlength'] added on line 20

File: src/components/number-input/number-input.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (altffyra) [reply src/components/number-input/number-input.vue]: image
changed to body 1 and body 2

File: src/components/number-input/number-input.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (altffyra) [reply src/components/number-input/number-input.vue]: removed restrictions and going on max length instead

File: src/components/number-input/number-input.vue (hunk 7 above)
Comment (altffyra) [reply src/components/number-input/number-input.vue:68]: removed label and space

**Review by saradalsman — APPROVED** [review]
Woho 🎉

**Review by adamfagerstrom — APPROVED** [review]
Awesome! Great work Christian!

**Pull request merged**